stubs/regex/regex/regex.pyi

```diff
@@ -0,0 +1,315 @@
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> _regex.Match[AnyStr] | None: ...
+@overload
+def match(
+    pattern: AnyStr,
+    string: AnyStr,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    partial: bool = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> _regex.Match[AnyStr] | None: ...
+@overload
+def match(
+    pattern: _regex.Pattern[AnyStr],
+    string: AnyStr,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    partial: bool = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> _regex.Match[AnyStr] | None: ...
+@overload
+def fullmatch(
+    pattern: AnyStr,
+    string: AnyStr,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    partial: bool = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> _regex.Match[AnyStr] | None: ...
+@overload
+def fullmatch(
+    pattern: _regex.Pattern[AnyStr],
+    string: AnyStr,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    partial: bool = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> _regex.Match[AnyStr] | None: ...
+@overload
+def split(
+    pattern: AnyStr,
+    string: AnyStr,
+    maxsplit: int = ...,
+    flags: int = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> list[AnyStr]: ...
+@overload
+def split(
+    pattern: _regex.Pattern[AnyStr],
+    string: AnyStr,
+    maxsplit: int = ...,
+    flags: int = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> list[AnyStr]: ...
+@overload
+def splititer(
+    pattern: AnyStr,
+    string: AnyStr,
+    maxsplit: int = ...,
+    flags: int = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> _regex.Splitter[AnyStr]: ...
+@overload
+def splititer(
+    pattern: _regex.Pattern[AnyStr],
+    string: AnyStr,
+    maxsplit: int = ...,
+    flags: int = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> _regex.Splitter[AnyStr]: ...
+@overload
+def findall(
+    pattern: AnyStr,
+    string: AnyStr,
+    flags: int = ...,
+    pos: int | None = ...,
+    endpos: int | None = ...,
+    overlapped: bool = ...,
+    concurrent: bool | None = ...,
+    timeout: int | None = ...,
+    ignore_unused: bool = ...,
+    **kwargs: Any,
+) -> list[AnyStr] | list[tuple[AnyStr, ...]]: ...
```

Comment:
This function returns one of `list[AnyStr]` or `list[tuple[AnyStr, ...]]` depending on if the pattern has sub-groups or not:
![image](https://user-images.githubusercontent.com/80432516/147619020-96e96a4a-5df8-4213-a133-e790d7cfb17b.png)

Given that the return type depends on the contents of the pattern rather than it's type I can't use an `overload` to handle this and I see in CONTRIBUTING that we shouldn't return a union of types:
https://github.com/python/mypy/issues/1693

Guess `list[Any]` is appropriate here then? (that's what `re.pyi` does)

Reply:
`list[Sequence[AnyStr]]` might cover both cases

Reply:
@AlexWaygood I tried a toy example and don't think `list[Sequence[AnyStr]]` covers `list[AnyStr]`:
![image](https://user-images.githubusercontent.com/80432516/147620858-d39ad2d2-4431-4600-9042-7f4709092ad4.png)

Reply:
Indeed, bytes are sequences of ints, not sequences of bytes.

Reply:
`list[Any]` it is, then.

Reply:
Although come to think of it, maybe `list[Sequence[Any]]` would work?

Reply:
I've updated to `list[Any]`